src/Evolution/Systems/Cce/AnalyticSolutions/TeukolskyWave.cpp

```diff
@@ -0,0 +1,316 @@
+          4.0 * square(extraction_radius_) * square(retarded_time) -
+          2.0 * square(duration_) * extraction_radius_ *
+              (extraction_radius_ + 3.0 * retarded_time));
+}
+
+double TeukolskyWave::pulse_profile_coefficient_b(const double time) const
+    noexcept {
+  const double retarded_time = time - extraction_radius_;
+  return 2.0 * amplitude_ * exp(-square(retarded_time) / square(duration_)) /
+         (pow<6>(duration_) * pow<5>(extraction_radius_)) *
+         (-3.0 * pow<6>(duration_) +
+          4.0 * pow<3>(extraction_radius_) * pow<3>(retarded_time) -
+          6.0 * square(duration_) * pow<2>(extraction_radius_) * retarded_time *
+              (extraction_radius_ + retarded_time) +
+          3.0 * pow<4>(duration_) * extraction_radius_ *
+              (extraction_radius_ + 2.0 * retarded_time));
+}
+
+double TeukolskyWave::pulse_profile_coefficient_c(const double time) const
+    noexcept {
+  const double retarded_time = time - extraction_radius_;
+  return 0.25 * amplitude_ * exp(-square(retarded_time) / square(duration_)) /
+         (pow<8>(duration_) * pow<5>(extraction_radius_)) *
+         (21.0 * pow<8>(duration_) +
+          16.0 * pow<4>(extraction_radius_) * pow<4>(retarded_time) -
+          16.0 * square(duration_) * pow<3>(extraction_radius_) *
+              square(retarded_time) *
+              (3.0 * extraction_radius_ + retarded_time) -
+          6.0 * pow<6>(duration_) * extraction_radius_ *
+              (3.0 * extraction_radius_ + 7.0 * retarded_time) +
+          12.0 * pow<4>(duration_) * square(extraction_radius_) *
+              (square(extraction_radius_) +
+               2.0 * extraction_radius_ * retarded_time +
+               3.0 * square(retarded_time)));
+}
+
+double TeukolskyWave::dr_pulse_profile_coefficient_a(const double time) const
+    noexcept {
+  const double retarded_time = time - extraction_radius_;
+  return -dt_pulse_profile_coefficient_a(time) -
+         9.0 * amplitude_ * exp(-square(retarded_time) / square(duration_)) /
+             (pow<4>(duration_) * pow<6>(extraction_radius_)) *
+             (5.0 * pow<4>(duration_) +
+              4.0 * square(extraction_radius_) * square(retarded_time) -
+              2.0 * square(duration_) * extraction_radius_ *
+                  (extraction_radius_ + 4.0 * retarded_time));
+}
+
+double TeukolskyWave::dr_pulse_profile_coefficient_b(const double time) const
+    noexcept {
+  const double retarded_time = time - extraction_radius_;
+  return -dt_pulse_profile_coefficient_b(time) +
+         2.0 * amplitude_ * exp(-square(retarded_time) / square(duration_)) /
+             (pow<6>(duration_ * extraction_radius_)) *
+             (15.0 * pow<6>(duration_) -
+              8.0 * pow<3>(extraction_radius_) * pow<3>(retarded_time) +
+              6.0 * square(duration_) * square(extraction_radius_) *
+                  retarded_time *
+                  (2.0 * extraction_radius_ + 3.0 * retarded_time) -
```

Comment:
the documentation has a `+` sign for the last term

Reply:
Checked through and it was the documentation in error this time -- I squashed the sign fix in. Thanks!